ns1/resource_record.go

```diff
@@ -377,6 +377,15 @@ func RecordCreate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 	if _, err := client.Records.Create(r); err != nil {
+		if err == ns1.ErrRecordExists && d.Get("OverwriteAllowed").(bool) {
```

Comment:
`d.Get` returns nil if the argument is unset. Depending on how we add to the schema, that shouldn't be an issue and/or we could use `getOk` ... we'll see how it shakes out.